### ISSUE TITLE
Validation failure with mongoose 5

### ISSUE THREAD

**Issue by jzwiesler** [vkarpov15/mongoose-unique-array#5]
I updated my mongoose version to 5.6.4 but now I get the following error with mongoose-unique-array 0.3.1:

`myModelvalidation failed: subDocument: this.getValue is not a function`

My Model is: 

```
const myModel = module.exports = new mongoose.Schema({
    _id: { type: String, required: true },
    baseUnitId: { type: String, required: true },
    subDocuments: { type: [subDocumentSchema], default: undefined },
}, {
    id: false
});
myModel.plugin(arrayUniquePlugin);

const subDocumentSchema= new mongoose.Schema({
    _id: { type: String, required: true, unique: true, sparse: true },
...
}, {
    id: false,
    autoIndex: true,
);
```

Is there anything I can do to fix this for mongoose 5.6.4?

**Comment by vkarpov15** [comment]
Released mongoose-unique-array 0.3.2 with a fix for this issue, upgrade and this issue should be fixed :+1:

**Comment by Fonger** [comment]
@vkarpov15  The fix actually doesn't work until #7 is merged.

**Comment by vkarpov15** [comment]
Released 0.3.3 with #7 , thanks @Fonger :+1:
